### DISCUSSION TITLE
Where is the directory for Presets or how can it be found?

### DISCUSSION THREAD

**Discussion by MicHaeL-MonStaR** [wwmm/easyeffects#2558] [Q&A]
I had already found a thread (from last year) that asked about exporting, which was answered with going to the directory. But even with the suggestions there, I still can't find the presets for the life of me. - I could find them when I tried the older PulseEffects on Ubuntu about half a year ago, but since I switched to Easy Effects when I moved to Pop!_OS later, I can't seem to find the individual preset-files. 

I've actually searched in all kinds of directories, also using search with terms like "pulse" (which did bring up a lot of results, but they seem to be for the general "pulse audio" tech), "easyeffects", just "easy", and even the names of the presets, but for some reason the individual preset-files still don't show up. - I don't know if they're not there (and hidden within some singular file), because it's a Flatpak, because the search in Pop!_OS fails to find them, or what have you.

It would be nice to make a backup before updating, because you never know when a piece of software resets itself. And I've made a lot of tiny tweaks to my main equalizer-settings (30 bands, so try and remember that). - The best I can do is either take screenshots for the lazy way or note everything down the hard way (and if needed, set everything manually again).

_While this would rather be for ideas/suggestions: I know that you can import presets, but for some reason you can't export them. It would be nice to have that ability as well, be it for each preset individually or to make a backup with all presets into a single file (or both ways, even better). - If not, at least a way to open the preset-directory to be able to grab them easily._

I'm hesitant to update to a new version without having a copy of the presets, or at least my main one, cause it was a long process to get it to where it is. - So if anyone could tell me where I could find the presets in Pop!_OS, that'd help a lot already.

Thanks for reading!

**Comment by wwmm** [comment]
When the installation is done through an native distribution package the presets can be found inside the user folder `~/.config/easyeffects/`. But when the installation is done through a Flatpak package the path is different. I thin it is something like `~/.var/app/com.github.wwmm.easyeffects`

**Reply by MicHaeL-MonStaR** [reply]
Those directories don't exist for me. `var` does, but `app` does not, so it ends there. - I did a search again and did find some things related to "easyeffects". - There is indeed a `com.github.wwmm.easyeffects`-folder, two in fact, but both are empty and one has a lock on it. - The former is in `run/user/1000/app` and the latter is in `run/user/1000/doc/by-app`.

Then there's one more folder called `app-flatpak-com.github.wwmm.easyeffects-3021.scope`, in some convoluted directory, which seems like the closest thing. But it has a bunch of files in that don't indicate anything about Presets. They start with words like `cgroup`, `cpu`, `io`, `memory`, and `pids`. But those seem like they're just the actual program-files.

Does one of these files hold the Presets by any chance? - Or am I even around where they should be?

**Reply by vchernin** [reply]
By default for flatpak installations `~/.var/app/com.github.wwmm.easyeffects` is used (as accessed from the host filesystem). You seem to be finding internal flatpak directories not meant to be viewed directly. Is it possible you configured flatpak to use a different directory to store app data?

You could try run `flatpak run --command=bash com.github.wwmm.easyeffects` and then enter `echo $XDG_CONFIG_HOME`. That will list the config directory used by the flatpak easyeffects app, which you should be able to visit from your host filesystem.

**Reply by MicHaeL-MonStaR** [reply]
I have changed nothing from the default of Pop!_OS as far as I know. I try not to change anything like that.

I've entered what you said and it does tell me the directory you described indeed. - However, it tells me `/home/name/.var/etc...`, but when I'm in the Home-folder, there is no folder with my name (or computer-name).
Actually, when I browse to the "Other Locations" (which is just showing all drives), and I click the drive of my Pop!_OS installation, I CAN find the name with the "computer-name", which is indeed in a folder called 'home', but I just end up in the Home-folder I described before.
There IS a `var`-folder (without the period) right on the Pop!_OS drive, parallel to the `home`-folder, but it doesn't hold an `app` folder.

I don't know what is going on. I could search other driver, but those have other installations on them, or just files without any OS or programs, and they're not even mounted, so I doubt they have it on them.

Is this just some kind of weird Pop!_OS-thing that makes it difficult to navigate?

**Reply by wwmm** [reply]
I think we will have to add some kind of export function to make things easier for people using Flatpak packages. I have never done it because usually it is just a matter of opening our folder with the system file browser or through a terminal. It usually is a straightforward process.

@MicHaeL-MonStaR a last resort to find the correct path would be to run EasyEffects in debug mode. First make sure you kill its process. Then start it as `G_MESSAGES_DEBUG=easyeffects flatpak run com.github.wwmm.easyeffects`. The next step is saving or creating a preset. The full path should be visible in the terminal.
